src/Curves/CurveOperations.jl

```diff
@@ -0,0 +1,226 @@
+    elseif curveType == "END_POINTS_LINE"
+        xStart = realArrayForKeyFromDictionary("xStart",crvDict)
+        xEnd   = realArrayForKeyFromDictionary("xEnd",crvDict)
+        x = zeros(Float64,3,2)
+        t = zeros(Float64,3)
+        for i = 1:3
+            t[i] = (i-1)/2.0
+        end
+
+        endPointsLineCurvePoints(xStart,xEnd,t,x)
+    elseif curveType == "CIRCULAR_ARC"
+        center     = realArrayForKeyFromDictionary("center",crvDict)
+        radius     = realForKeyFromDictionary("radius",crvDict)
+        startAngle = realForKeyFromDictionary("start angle",crvDict)
+        endAngle   = realForKeyFromDictionary("end angle",crvDict)
+        units      = crvDict["units"]
+
+        x = zeros(Float64,N+1,2)
+        t = zeros(Float64,N+1)
+        for i = 1:N+1
+            t[i] = (i-1)/N
+        end
+
+        arcCurvePoints(center,radius,startAngle,endAngle,units,t,x)
+    elseif curveType == "SPLINE_CURVE"
+        nKnots = intForKeyFromDictionary("nKnots",crvDict)
+        splineData = crvDict["SPLINE_DATA"]
+
+        M = max(N,nKnots*2)
+        x = zeros(Float64,M+1,2)
+        t = zeros(Float64,M+1)
+        for i = 1:M+1
+            t[i] = (i-1)/M
+        end
+
+        splineCurvePoints(nKnots,splineData,x)
+    end
+    return x
+end
+
+
+function chainPoints(chain::Array{Dict{String,Any}}, N::Int)
+
+    x = Any[]
+
+    for crvDict in chain
+        push!(x,curvePoints(crvDict,N))
+    end
+    return x
+end
+
+
+function curvePoint(crvDict::Dict{String,Any}, t::Float64)
+
+    curveType::String = crvDict["TYPE"]
+
+    if curveType == "PARAMETRIC_EQUATION_CURVE"
+        xEqn = crvDict["xEqn"]
+        yEqn = crvDict["yEqn"]
+        x = zeros(Float64,3)
+        peEquationCurvePoint(xEqn, yEqn, t, x)
+    elseif curveType == "END_POINTS_LINE"
+        xStart = realArrayForKeyFromDictionary("xStart",crvDict)
+        xEnd   = realArrayForKeyFromDictionary("xEnd",crvDict)
+        x = zeros(Float64,3)
+        endPointsLineCurvePoint(xStart,xEnd,t,x)
+    elseif curveType == "CIRCULAR_ARC"
+        center     = realArrayForKeyFromDictionary("center",crvDict)
+        radius     = realForKeyFromDictionary("radius",crvDict)
+        startAngle = realForKeyFromDictionary("start angle",crvDict)
+        endAngle   = realForKeyFromDictionary("end angle",crvDict)
+        units      = crvDict["units"]
+        x = zeros(Float64,3)
+        arcCurvePoint(center,radius,startAngle,endAngle,units,t,x)
+    elseif curveType == "SPLINE_CURVE"
+        nKnots = intForKeyFromDictionary("nKnots",crvDict)
+        splineData = crvDict["SPLINE_DATA"]
+        x = zeros(Float64,3)
+        splineCurvePoint(nKnots,splineData,t,x)
+    end
+    return x
+end
+
+
+function curvesMeet(firstCurve::Dict{String,Any}, secondCurve::Dict{String,Any})
+    xFirst  = curvePoint(firstCurve,1.0)
+    xSecond = curvePoint(secondCurve,0.0)
+    if maximum(abs.(xFirst - xSecond)) < 100*eps(Float64)
```

Comment:
```suggestion
function curvesMeet(firstCurve::Dict{String,Any}, secondCurve::Dict{String,Any}; tol=100*eps(Float64))
    xFirst  = curvePoint(firstCurve,1.0)
    xSecond = curvePoint(secondCurve,0.0)
    if maximum(abs.(xFirst - xSecond)) < tol
```

In case you want to let the user specify `tol` as a keyword argument, otherwise defaults to what you have here